eng/common/scripts/Package-Properties.ps1

```diff
@@ -129,19 +129,30 @@ function Get-PrPkgProperties([string]$InputDiffJson) {
             if ($shouldInclude) {
                 $packagesWithChanges += $pkg
 
-                if ($pkg.DependentPackages) {
-                    $dependentPackagesForInclusion += $pkg.DependentPackages
+                if ($pkg.CanaryPackages) {
+                    $dependentPackagesForInclusion += $pkg.CanaryPackages
                 }
             }
         }
     }
 
-    foreach ($addition in $dependentPackagesForInclusion) {
-        if ($lookup[$addition]) {
-            $packagesWithChanges += $lookup[$addition]
+    # there is a custom function to get dependent packages from a changed package set and diff
+    if ($DependentPackagesFromPackageSetFn -and (Test-Path "Function:$DependentPackagesFromPackageSetFn"))
+    {
+        $packagesWithChanges += &$DependentPackagesFromPackageSetFn $packagesWithChanges $diff
```

Comment:
This function should be implemented in `Language-Settings.ps1`. It accepts the pkgProps list for the detected packages that were included in the change, and the pscustomobject representation of the diff.json.

Reply:
Do you need to work with `$dependentPackagesForInclusion` and `$pkg.CanaryPackages` above if Canary packages above if the language overrides `"Get-${Language}-DependentPackagesFromPackageSet"`

Reply:
I don't understand what you're asking. We're giving the function A) the packages props that were changed (first arg) and B) we're handing you the package diff itself. I'm doing that for additional customization if you want it.

Reply:
I see. You're saying

> why not offer an override for this entire function

Because Wes didn't want me to, I already started there.

Reply:
Aha. You're asking if we want to interact with the canary packages at all if the function is defined. The answer is NO. I'll remove the add from the function above if the function is defined.